packages/samples/headless-commerce-react/src/components/instant-products/instant-products.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  InstantProducts as HeadlessInstantProducts,
+  Product,
+} from '@coveo/headless/commerce';
+import {useEffect, useState} from 'react';
+
+interface IInstantProductProps {
+  controller: HeadlessInstantProducts;
+  navigate: (pathName: string) => void;
+}
+
+export default function InstantProducts(props: IInstantProductProps) {
+  const {controller, navigate} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(
+    () => controller.subscribe(() => setState({...controller.state})),
```

Comment:
Why do we spread the state?
```suggestion
    () => controller.subscribe(() => setState(controller.state)),
```

Reply:
Oh yeah I just started from the result list component as a boilerplate, and for some reason I spread the state there...

No need to spread it :D

Reply:
For state objects its better to spread to make sure the nested properties changed are detected. [here](https://react.dev/reference/react/useState#examples-objects)

Reply:
OK so I apparently accidentally applied a best practice then hehe